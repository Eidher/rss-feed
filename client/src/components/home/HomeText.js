import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

class HomeText extends PureComponent {
  static propTypes = {
    mainText: PropTypes.string.isRequired,
    buttonText: PropTypes.string.isRequired
  };
  render() {
    return (
      <div className="col-lg-7 my-auto">
        <div className="content mx-auto">
          <h1 className="mb-5">{this.props.mainText}</h1>
          <Link to="/feed" className="btn btn-outline btn-xl">
            {this.props.buttonText}
          </Link>
        </div>
      </div>
    );
  }
}

export default HomeText;
