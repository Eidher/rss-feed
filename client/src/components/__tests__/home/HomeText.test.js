import React from "react";
import { shallow } from "enzyme";
import HomeText from "../../home/HomeText";

describe("<HomeText />", () => {
  const props = {
    mainText: "main text",
    buttonText: "button text samole"
  };
  const wrapper = shallow(<HomeText {...props} />);

  it("renders correctly.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has the right props ", () => {
    expect(wrapper.find("h1")).toHaveText(props.mainText);
    expect(wrapper.find("a")).toHaveText(props.buttonText);
  });
});
