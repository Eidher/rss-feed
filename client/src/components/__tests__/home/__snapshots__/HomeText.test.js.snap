// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeText /> renders correctly. 1`] = `
<div
  className="col-lg-7 my-auto"
>
  <div
    className="header-content mx-auto"
  >
    <h1
      className="mb-5"
    >
      main text
    </h1>
    <a
      className="btn btn-outline btn-xl js-scroll-trigger"
      href="#download"
    >
      button text samole
    </a>
  </div>
</div>
`;
